Xamarin.Forms.Core/Button.cs

```diff
@@ -240,6 +258,43 @@ void OnSourcePropertyChanging(ImageSource oldvalue, ImageSource newvalue)
 				oldvalue.SourceChanged -= OnSourceChanged;
 		}
 
+		static void BorderRadiusPropertyChanged(BindableObject bindable, object oldvalue, object newvalue)
+		{
+			if (newvalue == oldvalue)
+				return;
+
+			var val = (int)newvalue;
+			if (val == DefaultBorderRadius)
+				val = DefaultCornerRadius;
+
+			var oldVal = (int)bindable.GetValue(Button.CornerRadiusProperty);
+
+			if (oldVal == val)
+				return;
+
+			bindable.SetValue(Button.CornerRadiusProperty, val);
+		}
+
+		static void CornerRadiusPropertyChanged(BindableObject bindable, object oldvalue, object newvalue)
+		{
+			if (newvalue == oldvalue)
+				return;
+
+			var val = (int)newvalue;
+			if (val == DefaultCornerRadius)
+				val = DefaultBorderRadius;
+
+#pragma warning disable 0618 // retain until BorderRadiusProperty removed
+			var oldVal = (int)bindable.GetValue(Button.BorderRadiusProperty);
+#pragma warning restore
+
+			if (oldVal == val)
+				return;
+
+#pragma warning disable 0618 // retain until BorderRadiusProperty removed
+			bindable.SetValue(Button.BorderRadiusProperty, val);
+#pragma warning restore
+		}
```

Comment:
I'm trying to think about all the things that could go wrong here:
- binding => should be fine, unless user binds to the 2 properties and has a poor INPC implementation in VM
- clearvalue => could you please add a unit test for button making sure that when a property is Cleared (using `ClearValue` both are actually cleared ? i.e. IsSet() == false; ?

Reply:
Unit tests added! However, it looks like we don't clear the context when we call `ClearValue`, so `IsSet` returns true in that case. Added a test to `BindableObject` to show that it's not specific to this case.

Reply:
😢